src/main/java/build/buildfarm/common/ContentAddressableStorage.java

```diff
@@ -0,0 +1,37 @@
+package build.buildfarm.common;
+
+import com.google.devtools.remoteexecution.v1test.Digest;
+import com.google.protobuf.ByteString;
+
+public interface ContentAddressableStorage {
```

Comment:
What about thread-safety?

Reply:
I'll add the thread safety annotations to the relevant members